Restyling of Files plugin

pmknutsen opened the pull request:
Darkened text color in frame (file list)
Changed hover effects
Emphasize folders further with neon-green color
Dropdown menu and hover effects more consisent with rest of UI
Subtle gradient in plugin headers
Subtle vertical border on sidebar

![screenshot from 2016-01-21 21 59 59](https://cloud.githubusercontent.com/assets/1565499/12503892/7145762a-c08d-11e5-8361-673b2ab5df87.png)
![screenshot from 2016-01-21 21 58 28](https://cloud.githubusercontent.com/assets/1565499/12503895/7360d9d6-c08d-11e5-917e-c9e7a5e3ca14.png)
![screenshot from 2016-01-21 21 58 44](https://cloud.githubusercontent.com/assets/1565499/12503897/76e85cb4-c08d-11e5-9ea4-17209e6cab1f.png)

pmknutsen committed:
Restyling of Files plugin
Dropdown menu and hover effects more consisent with rest of UI
Subtle gradient in plugin headers
Subtle vertical border on sidebar

DavidVorick commented:
While you are at it, can you find some alternative to the edit button? I'm guessing double-click doesn't work in the UI? The edit button looks pretty bad, especially next to every single file.

Maybe have a right-click menu on each file?

DavidVorick commented:
(thanks as always for contributing)

Mingling94 commented on plugins/Files/css/files.css:
```diff
@@ -121,8 +138,10 @@ hr {
+	background-color: #ECECEC;
+}
 .file.anchor {
-	border: 1px solid #00CBA0;
```
Why get rid of the anchor selection box? I thought it was neat and made the effect of shift+click selections explicit. If you are getting rid of it, also delete lines 124 & 126 because they're unneeded clutter.

Mingling94 commented on plugins/Files/index.html:
```diff
@@ -30,41 +30,51 @@
-							<span class='fa-stack'>
-								<i class='fa fa-folder-o fa-stack-2x'></i>
-								<i class='fa fa-plus fa-stack-1x'></i>
+							<span class="icon">
```
Nice! I suppose this aligns the icons ^_^

Mingling94 commented:
@DavidVorick Originally the behavior was:
single-click the name -> edit name (This is how Windows does it)

Then @lukechampine suggested in #175 and I implemented:
single-click button -> edit name (Google Drive & Ubuntu-Gnome have it in context menu, but I haven't experimented with context menu options specific to the plugins yet. It may be non-trivial)
double-click files -> download (save dialog pops up)

Double-click has always worked, as evidenced by double-clicking folders to navigate inside them

Mingling94 commented:
I think the green on the folders and dropdown selection are unnecessary. The green coloring so far is used for selections (as in the side bar) and emphasis (as in the capsule and for `button:hover` in the header of a plugin).

The green in the dropdown and folder icon are neither of these. Perhaps to emphasize the file/folder distinction, you could add another column, `Type`, like how Ubuntu-Gnome does it:
![selection_007](https://cloud.githubusercontent.com/assets/4606838/12516562/1ee4a576-c0fc-11e5-8d50-37cf93d013c8.png)

DavidVorick commented:
Or maybe a sia-purple could be used? I like the idea of coloring the folders differently, but it sounds like green was the wrong choice.

pmknutsen committed:
New Type column in file browser
Removed folder coloring
Removed border-bottom on each item to match OS file browsers
Reduced contrast of hor div in dropdown

pmknutsen commented:
I agree the color choice was poor; the neon green should be reserved for selected or high-lighted items. In case you didn't notice, I also used the green when hovering over directory names (see screenshot below; note pointer isn't seen).

I removed the folder color and added a Type column, which I think is a good idea.

![screenshot from 2016-01-22 08 56 00](https://cloud.githubusercontent.com/assets/1565499/12517134/20c4ee16-c0e6-11e5-913f-846b35d22700.png)

![screenshot from 2016-01-22 08 54 42](https://cloud.githubusercontent.com/assets/1565499/12517082/d4cb46e0-c0e5-11e5-8e3f-d538ab837b40.png)

pmknutsen commented:
This how it would look with purple. I think it clashes a bit with the rest.

![screenshot from 2016-01-22 09 01 30](https://cloud.githubusercontent.com/assets/1565499/12517277/cf9652a4-c0e6-11e5-897e-a826f63b2b52.png)

lukechampine commented:
I think the current combination of 1) slightly larger icon, 2) `/` after name, and 3) type "Folder" is enough to make folders distinguishable from files. A separate color is a sensible approach to try, but it seems difficult to pick the proper color without it looking gaudy. I think it's ok for now.

DavidVorick commented:
Are you only seeing 17 active hosts?

Jinglan commented:
I think that one helpful differentiation btwn folders and files would be this:
![archive142](https://cloud.githubusercontent.com/assets/9970082/12519575/ac22afda-c10d-11e5-8d54-fe42a28a0d2c.png)
![text70](https://cloud.githubusercontent.com/assets/9970082/12519577/ae19ac4e-c10d-11e5-8c5d-ffc68119b5f5.png)

Folder icon is dark filled, file icon is white filled.

DavidVorick commented:
that looks good to me

pmknutsen commented:
@Jinglan Agreed. That makes it clearer.

![screenshot from 2016-01-22 11 03 28](https://cloud.githubusercontent.com/assets/1565499/12520172/df4ada10-c0f7-11e5-86cc-2907f545ebb7.png)

lukechampine commented:
excellent!

DavidVorick commented:
I feel like the 'type' column is not needed, especially with the dark/light distinction between folders and files. I think the change is good.

I still cringe when I look at the edit-pencils. That doesn't need to be fixed in this PR though.

lukechampine commented:
agreed. We can consider readding "type" when it would add more information, like in Ming's screenshot above (though this may not be currently possible; .sia files don't contain that metadata).
The edit-pencils should be moved to a single button, alongside the others at the top.
btw, there's a bug related to renaming folders: siad doesn't support renaming folders, just files. So really when you rename a folder, the UI should actually be renaming all the files within that folder. Alternatively, we could make this siad's job.
Supporting drag-and-drop to move files in and out of folders would also be a great feature to have.

pmknutsen commented:
@lukechampine I removed Type column. Context menu won't be a problem for renaming a file etc, but there is a conflict with the window context-menu when right-clicking. @Mingling94 How to prevent the global context menu from showing when a specific element is clicked?

lukechampine commented:
It may make more sense to hold off on right-clicking behavior for now. If we want to add it, I think we'd also have to add support for all the usual actions (delete/download/rename) as well as right-clicking with multiple files selected.
on that subject, I think if/when we add right-click actions, they shouldn't replace the actions already visible in the top bar. The user may not be aware that the right-click menu exists.

DavidVorick commented:
siad should be the one that supports renaming folders.

Mingling94 commented:
@lukechampine
- Renaming a folder in the UI does just rename all the files within said folder. I've checked that this feature works with at least one file like: `Folder/File` -> `RenamedFolder/File`
- Also a rename button wouldn't really make sense as a top bar button because it doesn't apply to many items.
- Drag and drop would be cool

@pmknutsen 
- I wouldn't know exactly. Check out: https://github.com/NebulousLabs/Sia-UI/blob/master/index.html#L60 Since an event is what fires the CM, and it clearly bubbles up from webviews to the general index.html, you'd have to change these lines to only fire a global CM if the webview itself doesn't have a more specific one. Though like luke said, it can be saved for another PR.

lukechampine commented:
Whoops, correct you are. Not sure why I was mistaken about that. Maybe I was thinking of renaming empty directories, since siad doesn't understand them.

Good point on the rename button. It could simply be grayed out when multiple files were selected though.
Looking at other directory interfaces, they seem to get by without a rename button just fine. I think at this point the user expects two methods for renaming:
- Right click and select "Rename"
- Click once to highlight, wait for double-click timeout, then click again

Also, most UIs will highlight the filename when renaming. Might be a nice touch.

pmknutsen committed:
Pencil rename icon moved to top toolbar
Fixed bug with file toolbar not updating when directory is changed
White file icons
Styling of selected file/folder elements
Row highlighted as selected during renaming

pmknutsen commented:
Current version after commit 48cfd0c looks like this:

![screenshot from 2016-01-23 15 51 01](https://cloud.githubusercontent.com/assets/1565499/12533587/642d4ce4-c1e9-11e5-9e8c-d5237d8bfe0d.png)

@lukechampine The file row remains highlighted when renaming. Auto-selecting the entire filename would be nice too.

Rename icon simply does nothing if multiple files are selected. Grayed out or not showing at all would be better.

lukechampine commented:
nice. Looks like the build failed due to some frivolous syntax errors. Once those are cleaned up, I think this should be merged. Further changes can be made in a new PR.

pmknutsen committed:
Fix syntax error

pmknutsen commented:
@lukechampine Fixed the syntax errors but Travis build still fails. I don't understand the error. Could you have a look plz?

pmknutsen committed:
Fixed filename reference that prevented renaming in subdirs.

lukechampine commented:
just some nondeterminism. I've rerun the build. Thanks!
